webpack/dev.config.js

```diff
@@ -6,73 +6,96 @@ import webpack from 'webpack';
 import writeStats from './utils/write-stats';
 import startKoa from './utils/start-koa';
 
-const WEBPACK_HOST = 'localhost';
+const PROTOCOL = (process.env.C9_HOSTNAME) ? 'https' : 'http';
+const HOST = process.env.C9_HOSTNAME || require('os').hostname();
```

Comment:
On my mac `require('os').hostname();` is equal to: `retina.local`

You should use `localhost` directly on other then Cloud9 ENV